### ISSUE TITLE
examples/summarize_rlhf/reward_model eval got 50% acc

### ISSUE THREAD

**Issue by scarydemon2** [CarperAI/trlx#338]
### 🐛 Describe the bug

I used 'gpt2-xl' to train this reward model without deepspeed config used, and no extra code was modified. 
For every 200 steps, I did an evaluation, and the ACC was almost always around 50%, equal to a random guess. 
![image](https://user-images.githubusercontent.com/33051998/221738481-9f88c4bd-e8b5-47a2-ae2a-92c07d28b940.png)

The loss value reduces normally.
![image](https://user-images.githubusercontent.com/33051998/221738124-780298b6-ef14-4ef7-9731-4e8540d1a07b.png)

my training args shows as follow: 
![image](https://user-images.githubusercontent.com/33051998/221738336-b32cf0f4-d90a-4c40-a7e8-d662d8f9e76c.png)

So what could possibly go wrong？ If there anyone reproduces the reward model successfully

### Which trlX version are you using?

0.5.0

### Additional system and package information

torch:1.13.1 transformers:4.26.1  accelerate 0.16.0

**Comment by PhungVanDuy** [comment]
> ### 🐛 Describe the bug
> I used 'gpt2-xl' to train this reward model without deepspeed config used, and no extra code was modified. For every 200 steps, I did an evaluation, and the ACC was almost always around 50%, equal to a random guess. ![image](https://user-images.githubusercontent.com/33051998/221738481-9f88c4bd-e8b5-47a2-ae2a-92c07d28b940.png)
> 
> The loss value reduces normally. ![image](https://user-images.githubusercontent.com/33051998/221738124-780298b6-ef14-4ef7-9731-4e8540d1a07b.png)
> 
> my training args shows as follow: ![image](https://user-images.githubusercontent.com/33051998/221738336-b32cf0f4-d90a-4c40-a7e8-d662d8f9e76c.png)
> 
> So what could possibly go wrong？ If there anyone reproduces the reward model successfully
> 
> ### Which trlX version are you using?
> 0.5.0
> 
> ### Additional system and package information
> torch:1.13.1 transformers:4.26.1 accelerate 0.16.0

Did you tried to fine-tune SFT for GPT-XL before trained reward model?

**Comment by scarydemon2** [comment]
> > ### 🐛 Describe the bug
> > I used 'gpt2-xl' to train this reward model without deepspeed config used, and no extra code was modified. For every 200 steps, I did an evaluation, and the ACC was almost always around 50%, equal to a random guess. ![image](https://user-images.githubusercontent.com/33051998/221738481-9f88c4bd-e8b5-47a2-ae2a-92c07d28b940.png)
> > The loss value reduces normally. ![image](https://user-images.githubusercontent.com/33051998/221738124-780298b6-ef14-4ef7-9731-4e8540d1a07b.png)
> > my training args shows as follow: ![image](https://user-images.githubusercontent.com/33051998/221738336-b32cf0f4-d90a-4c40-a7e8-d662d8f9e76c.png)
> > So what could possibly go wrong？ If there anyone reproduces the reward model successfully
> > ### Which trlX version are you using?
> > 0.5.0
> > ### Additional system and package information
> > torch:1.13.1 transformers:4.26.1 accelerate 0.16.0
> 
> Did you try to fine-tune SFT for GPT-XL before training the reward model?

 Not yet, I have thought SFT is fine-tuned for better training the PPO model. But I think you are right I should train SFT first, because pre-trained gpt2 may just generate meanless sentences when seeing the given prompt.  I will try to train the SFT first.

**Comment by PhungVanDuy** [comment]
> > > ### 🐛 Describe the bug
> > > I used 'gpt2-xl' to train this reward model without deepspeed config used, and no extra code was modified. For every 200 steps, I did an evaluation, and the ACC was almost always around 50%, equal to a random guess. ![image](https://user-images.githubusercontent.com/33051998/221738481-9f88c4bd-e8b5-47a2-ae2a-92c07d28b940.png)
> > > The loss value reduces normally. ![image](https://user-images.githubusercontent.com/33051998/221738124-780298b6-ef14-4ef7-9731-4e8540d1a07b.png)
> > > my training args shows as follow: ![image](https://user-images.githubusercontent.com/33051998/221738336-b32cf0f4-d90a-4c40-a7e8-d662d8f9e76c.png)
> > > So what could possibly go wrong？ If there anyone reproduces the reward model successfully
> > > ### Which trlX version are you using?
> > > 0.5.0
> > > ### Additional system and package information
> > > torch:1.13.1 transformers:4.26.1 accelerate 0.16.0
> > 
> > 
> > Did you try to fine-tune SFT for GPT-XL before training the reward model?
> 
> Not yet, I have thought SFT is fine-tuned for better training the PPO model. But I think you are right I should train SFT first, because pre-trained gpt2 may just generate meanless sentences when seeing the given prompt. I will try to train the SFT first.

Thank you, I also suggest that you use this [repo](https://github.com/Dahoas/reward-modeling), to train your reward model with more options to configure the loss (sparse, dense), easier configuration,...

**Comment by scarydemon2** [comment]
> > > > ### 🐛 Describe the bug
> > > > I used 'gpt2-xl' to train this reward model without deepspeed config used, and no extra code was modified. For every 200 steps, I did an evaluation, and the ACC was almost always around 50%, equal to a random guess. ![image](https://user-images.githubusercontent.com/33051998/221738481-9f88c4bd-e8b5-47a2-ae2a-92c07d28b940.png)
> > > > The loss value reduces normally. ![image](https://user-images.githubusercontent.com/33051998/221738124-780298b6-ef14-4ef7-9731-4e8540d1a07b.png)
> > > > my training args shows as follow: ![image](https://user-images.githubusercontent.com/33051998/221738336-b32cf0f4-d90a-4c40-a7e8-d662d8f9e76c.png)
> > > > So what could possibly go wrong？ If there anyone reproduces the reward model successfully
> > > > ### Which trlX version are you using?
> > > > 0.5.0
> > > > ### Additional system and package information
> > > > torch:1.13.1 transformers:4.26.1 accelerate 0.16.0
> > > 
> > > 
> > > Did you try to fine-tune SFT for GPT-XL before training the reward model?
> > 
> > 
> > Not yet, I have thought SFT is fine-tuned for better training the PPO model. But I think you are right I should train SFT first, because pre-trained gpt2 may just generate meanless sentences when seeing the given prompt. I will try to train the SFT first.
> 
> Thank you, I also suggest that you use this [repo](https://github.com/Dahoas/reward-modeling), to train your reward model with more options to configure the loss (sparse, dense), easier configuration,...

haha, Thanks for your advice. I'll try it.
